File: Ahorn/triggers/DelayedKillTrigger.jl
module VectorHelperDelayedKillTrigger

using ..Ahorn, Maple

@mapdef Trigger "VectorHelper/DelayedKillTrigger" DelayedKillTrigger(
	x::Integer, y::Integer, width::Integer=Maple.defaultTriggerWidth, height::Integer=Maple.defaultTriggerHeight,
	delay::Number=1.0, resetDelayOnLeave::Bool=true, resetDelayIfInvincible::Bool=false,
	ignoreInvincibleAssist::Bool=false, addToDeathCount::Bool=false
)

const placements = Ahorn.PlacementDict(
	"Kill Trigger (Delayed) (Vector Helper)" => Ahorn.EntityPlacement(
		DelayedKillTrigger,
		"rectangle"
	)
)

Ahorn.editingOrder(entity::DelayedKillTrigger) = String[
	"x", "y",
	"width", "height",
	"delay", "ignoreInvincibleAssist", "addToDeathCount",
	"resetDelayOnLeave", "resetDelayIfInvincible"	
]

end

File: Loenn/triggers/DelayedKillTrigger.lua
--VectorHelper Delayed Kill trigger (Loenn Integration)
--Written by KthVeg

local DelayKillTrigger = {}

DelayKillTrigger.name = "VectorHelper/DelayedKillTrigger"
DelayKillTrigger.fieldOrder = {
    "x", "y",
    "width", "height",
    "delay", "ignoreInvincibleAssist",
    "addToDeathCount", "resetDelayOnLeave"
}
DelayKillTrigger.placements = {
    {
        name = "Kill Trigger (Delayed)",
        data = {
            delay = 1.0,
            resetDelayOnLeave = true,
            resetDelayIfInvincible = false,
            addToDeathCount = false,
            ignoreInvincibleAssist = false
        }
    }
}

return DelayKillTrigger